packages/ui/src/routes/networks/NetworkFormPage.tsx

```diff
@@ -171,6 +192,11 @@ const NetworkFormPage = ({
         symbol: watchCurrencySymbol,
     } = watchedFields
 
+    useEffect(() => {
+        const isDefaultRPC = !!defaultRpcUrl && watchRPCUrl === defaultRpcUrl
```

Comment:
No need to use `useState` and `useEffect` to get the job done.  You can just use this definition to know whether the user is using the default or not.

```
const isDefaultRPC = !!defaultRpcUrl && watchRPCUrl === defaultRpcUrl
```